lib/trento_web/controllers/v1/suma_credentials_controller.ex

```diff
@@ -25,6 +27,33 @@ defmodule TrentoWeb.V1.SUMACredentialsController do
     end
   end
 
+  operation :create,
+    summary: "Saves the user settings",
+    tags: ["Platform"],
+    description: "Saves user settings for SUSE Manager",
+    request_body:
+      {"SUMACredentialsRequest", "application/json", SUMACredentials.SUMACredentialsRequest},
+    responses: [
+      created: {"Settings saved successfully", "application/json", SUMACredentials.Settings},
+      unprocessable_entity: UnprocessableEntity.response()
+    ]
+
+  @spec create(Plug.Conn.t(), any) :: Plug.Conn.t()
+  def create(%{body_params: body_params} = conn, _) do
+    with {:ok, saved_settings} <-
+           body_params |> Map.from_struct() |> SoftwareUpdates.save_settings() do
```

Comment:
Nitpick: just be careful with the pipes in with statements, they can become pretty unreadable fast (personal PTSD moment with that)